src/btree/bt_curnext.c

```diff
@@ -54,8 +55,10 @@ __cursor_fix_append_next(WT_CURSOR_BTREE *cbt, bool newpage)
 	 * insert is aborted, we simply return zero (empty), regardless of
 	 * whether we are at the end of the data.
 	 */
-	if (cbt->recno < WT_INSERT_RECNO(cbt->ins) ||
-	    (upd = __wt_txn_read(session, cbt->ins->upd)) == NULL) {
+	__wt_txn_read(session, cbt->ins->upd, &visibility, &upd);
+	if (visibility == WT_VISIBLE_PREPARE)
+		return (WT_PREPARE_CONFLICT);
+	if (cbt->recno < WT_INSERT_RECNO(cbt->ins) || upd == NULL) {
```

Comment:
Think about whether switching the order of the comparison and visibility check matters for correctness or performance.